packages/dds/tree/src/feature-libraries/forest-proxy/forestProxy.ts

```diff
@@ -0,0 +1,128 @@
+		if (!this._children.has(key)) {
+			const proxy = proxify(this.forest, this.cursor);
+			this._children.set(key, proxy);
+		}
+		return this._children.get(key);
+	}
+}
+
+const tryMoveDown = (cursor: ITreeSubscriptionCursor, key: FieldKey): TreeNavigationResult => {
+	const result = cursor.down(key, 0);
+	if (result === TreeNavigationResult.NotFound) {
+		// maybe an array?
+		return cursor.down(EmptyKey, Number(key));
+	}
+	return result;
+};
+
+const handler: ProxyHandler<TargetForest> = {
+	get: (target: TargetForest, key: string | symbol): any => {
+		if (typeof key === "symbol") {
+			return Reflect.get(target, key);
+		}
+		const result = tryMoveDown(target.cursor, key as FieldKey);
+		if (result === TreeNavigationResult.NotFound) {
+			return Reflect.get(target, key);
+		}
+		const value = target.cursor.value === undefined
+			? target.getChildProxy(key)
+			: target.cursor.value;
+		target.cursor.up();
+		return value;
+	},
+	set: (target: TargetForest, key: string | symbol, value: any): boolean => {
+		throw new Error("Not implemented.");
+	},
+	has: (target: TargetForest, key: string | symbol): boolean => {
+		if (key === proxySymbol) {
+			return true;
+		}
+		const result = tryMoveDown(target.cursor, key as FieldKey);
+		if (result === TreeNavigationResult.Ok) {
+			target.cursor.up();
+			return true;
+		}
+		return false;
+	},
+	ownKeys(target: TargetForest) {
+		return target.cursor.keys as string[];
+	},
+	getOwnPropertyDescriptor(target: TargetForest, key: string | symbol) {
+		if (key === proxySymbol) {
+			return { configurable: true, enumerable: true, value: key, writable: false };
+		}
+		const result = tryMoveDown(target.cursor, key as FieldKey);
+		if (result === TreeNavigationResult.Ok) {
+			const value = target.cursor.value === undefined
+				? target.getChildProxy(key as string)
+				: target.cursor.value;
+			const descriptor = {
+				configurable: true,
+				enumerable: true,
+				value,
+				writable: true,
+			};
+			target.cursor.up();
+			return descriptor;
+		}
+		return undefined;
+	},
+};
+
+const proxify = (forest: IEditableForest, cursor?: ITreeSubscriptionCursor) => {
+	// A TargetForest constructor unconditionally allocates a new cursor or forks the one if exists.
+	// Keep in mind that they must be cleared at some point (e.g. before writing to the forest).
+	// It does not modify the cursor.
+	const target = new TargetForest(forest, cursor);
+	const proxy = new Proxy(target, handler);
+	Object.defineProperty(proxy, proxySymbol, {
+		enumerable: false,
+		configurable: true,
+		writable: false,
+		value: proxySymbol,
+	});
+
+	return proxy;
+};
+
+/**
+ * Proxify a Forest to showcase basic interaction scenarios.
+ * This function forwards Forest to be proxified to minimize exported signature.
+ * It is the only package level export for forestProxy.
+ * @returns a proxy wrapping the given {@link IEditableForest}.
+ */
+export const proxifyForest = (forest: IEditableForest) => proxify(forest);
```

Comment:
the return type of this function is implicitly a type which is not exported  (TargetForest), not documented, and hard to figure out from looking at the code (you have to read the implementation, then follow that to proxify and read its implementation). We mostly avoid implicit return types in our code.

This also breaks the build (API extractor requires package exported APIs to only use package exported types).

This code should explicitly state a return type, which should be an exported and documented type.